tests/framework/BaseYiiTest.php

```diff
@@ -60,4 +62,26 @@ public function testPowered()
     {
         $this->assertTrue(is_string(Yii::powered()));
     }
+
+    public function testCreateObjectCallable()
+    {
+        Yii::$container = new Container();
+
+        // Test passing in of normal params combined with DI params.
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a) {
+            return $a === 'a';
+        }, ['a']));
+
+
+        $singer = new Singer();
+        $singer->firstName = 'Bob';
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a) {
+            return $singer->firstName === 'Bob';
+        }, [$singer, 'a']));
+
+
+        $this->assertTrue(Yii::createObject(function(Singer $singer, $a = 3) {
+            return true;
+        }));
```

Comment:
The code inside my closure will not run if the argument is not of the correct type; so no matter what return value I specify, it will not be returned... So returning true is as good a value as any..